app/models/user.rb

```diff
@@ -9,9 +9,67 @@ class User < ApplicationRecord
   validates :first_name, :last_name, :username, presence: true
   validates :username, uniqueness: true
 
-  def data
-    User.all.as_json.find do |user|
-      user['username'] == username
+  has_many :active_friendships, foreign_key: :active_friend_id, dependent: :destroy, class_name: 'Friendship'
+  has_many :passive_friendships, foreign_key: :passive_friend_id, dependent: :destroy, class_name: 'Friendship'
+  has_many :active_friends, through: :passive_friendships, source: :active_friend, dependent: :destroy
```

Comment:
Is it used anywhere?

Reply:
Oh haha nice catch! Ok in relation to the `friends` method I originally used the "old way" (using arrays) so I initially needed these two associations. Now removed them here and their corresponding specs.